### PR TITLE
verify stardust requirement in canPowerUp

### PR DESCRIPTION
**Pull request by AbandonedCart**
This verifies the user can actually "afford" to perform the power-up

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
rebase please from the latest commit to fix build error.

**Comment by LoungeKatt** [comment]
Relax, the commit came 20 minutes after the PR.

File: library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java [hunk 1]
```diff
@@ -170,9 +172,12 @@ public Result transferPokemon() throws LoginFailedException, RemoteServerExcepti
 	 * @return the boolean
 	 * @throws LoginFailedException  the login failed exception
 	 * @throws RemoteServerException the remote server exception
+	 * @throws InvalidCurrencyException the currency has exception
 	 */
-	public boolean canPowerUp() throws LoginFailedException, RemoteServerException {
-		return getCandy() >= getCandyCostsForPowerup();
+	public boolean canPowerUp() throws LoginFailedException, RemoteServerException, InvalidCurrencyException {
+		return getCandy() >= getCandyCostsForPowerup()
```
Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: get currency don´t throw a InvalidCurrencyException

Comment (AbandonedCart) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: Please read the entire change. `api.getPlayerProfile().getCurrency(PlayerProfile.Currency.STARDUST)` does.

Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: http://imgur.com/a/XWQFb

Comment (AbandonedCart) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: Take out the exception and compile it. You'll get an error. Why do you think both PRs handled that exception in one way or another? If you feel there shouldn't be an exception, then fix the original class to stop passing it to the reference. That isn't within the scope of this PR.

Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: There is no exception in the original class.

Comment (FabianTerhorst) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: https://github.com/Grover-c13/PokeGOAPI-Java/blob/Development/library/src/main/java/com/pokegoapi/api/player/PlayerProfile.java#L211

Comment (mgorski-mg) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: But it was an exception. I already fix it in last PR. 
@LoungeKatt pull last changes from Development and eveything should work without exceptions.

Comment (AbandonedCart) [inline library/src/main/java/com/pokegoapi/api/pokemon/Pokemon.java]: @FabianTerhorst The exception was removed 2 minutes before your comment. Luckily, @mgorski-zlatan mentioned that a recent change has modified the PR without presenting an error. I am not sure why there had to be this elaborate claim a commit from a day ago did not account for a change 45 minutes ago without simply referencing the change.

**Pull request merged**